src/Source/FileSource.js

```diff
@@ -169,7 +169,9 @@ class FileSource extends Source {
             if (data.extent) {
                 this.extent = data.extent.clone();
                 // Transform local extent to data.crs projection.
-                this.extent.applyMatrix4(data.matrixWorld);
+                if (this.extent.crs == data.crs) {
```

Comment:
As you relevantly suggested, maybe you could add a comment on the fact that both crs are different only when `data.crs` is 4978, and that if this is the case, applying the transformation to the extent would result in an irrelevant extent (since 4978 is a 3d crs).

Reply:
added